tests/Microsoft.Identity.Test.Performance/Program.cs

```diff
@@ -19,33 +21,40 @@ public static void Main(string[] args)
 
             try
             {
-                BenchmarkSwitcher.FromTypes(new[] {
+                var results = BenchmarkSwitcher.FromTypes(new[] {
                     typeof(AcquireTokenForClientCacheTests),
                     typeof(AcquireTokenForOboCacheTests),
                     typeof(TokenCacheTests),
                     typeof(AcquireTokenNoCacheTests),
-            }).RunAll(
+                }).RunAll(
 #if DEBUG
-                    new DebugInProcessConfig()
-                        .WithOptions(ConfigOptions.DontOverwriteResults) // When running manually locally
+                    new DebugInProcessConfig() // Allows debugging into benchmarks
+                        .WithOptions(ConfigOptions.DontOverwriteResults) // When debugging locally
 #else
                     DefaultConfig.Instance
+                        .AddJob(Job.Default.WithId("Job-PerfTests"))
 #endif
                         .WithOptions(ConfigOptions.DisableLogFile)
                         .WithOptions(ConfigOptions.StopOnFirstError)
-                        //.WithOptions(ConfigOptions.JoinSummary)
+                        //.WithOptions(ConfigOptions.JoinSummary) // Should be commented for Benchmark GitHub Action to work.
                         .WithOrderer(new DefaultOrderer(SummaryOrderPolicy.Method))
                         .HideColumns(Column.UnrollFactor, Column.Type, Column.InvocationCount, Column.Error, Column.StdDev, Column.Median, Column.Job)
                         .AddDiagnoser(MemoryDiagnoser.Default) // https://benchmarkdotnet.org/articles/configs/diagnosers.html
                         //.AddDiagnoser(new EtwProfiler()) // https://adamsitnik.com/ETW-Profiler/
-                .AddJob(
-                    Job.Default
-                        .WithId("Job-PerfTests")), args);
+                 , args);
+
+                // If no tests ran for whatever reason, throw an exception to break the build.
+                Summary summary = results?.FirstOrDefault();
+                BenchmarkReport report = summary?.Reports.FirstOrDefault();
+                if (summary == null || report == null || !report.Success)
```

Comment:
what do the results look like?

Reply:
It's everything that's printed out to the console during the run - collections of benchmark cases, all the metrics, environment info, settings, etc.